### PR TITLE
[linux-port] Add documentation file for Linux/macOS port.

### PR DESCRIPTION
**Pull request by ehsannas**

### CODE REVIEW

File: docs/DxcOnUnix.rst [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+Current Status
+==============
+
+Up and Running
+--------------
+We have currently reached the point where we can successfully build and run DXC
+on Linux and macOS. Code generation works for both DXIL and SPIR-V, and we are
+also able to fully run the SPIR-V CodeGen test suite on these platforms.
+
```
Comment (antiagainst) [inline docs/DxcOnUnix.rst]: I'd suggest to add a few words to declare that this is experimental right now. How we implement certain things may change without considering back portability.

File: docs/DxcOnUnix.rst [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+* dxr
+* dxv
+* dxlib-sample
+
+Moreover, since the HLSL CodeGen tests were originally written with Windows in
+mind, they require the Windows-specific `TAEF Framework <https://docs.microsoft.com/en-us/windows-hardware/drivers/taef/>`_
+to run. Therefore we are not able to compile/run these tests on non-Windows
+platforms. Note that it is only the testing infrastructure that has this
+limitation, and DXIL CodeGen works as expected by running the DXC executable.
+
```
Comment (antiagainst) [inline docs/DxcOnUnix.rst]: We also need to mention that the shared library of version number is stuck with 3.7 right now. That's something we need to fix.

**Comment by hekota** [comment]
👍

File: docs/DxcOnUnix.rst (hunk 1 above)
Comment (ehsannas) [reply docs/DxcOnUnix.rst]: Done.

File: docs/DxcOnUnix.rst (hunk 2 above)
Comment (ehsannas) [reply docs/DxcOnUnix.rst]: Added to the ``Known Issues`` section.

**Pull request merged**